CFC_DataCollector/trip_analysis/kmedoid.py

```diff
@@ -0,0 +1,122 @@
+import random
+import sys
+
+#code based on K_medoid_2.py in CFC_WebApp/main
```

Comment:
I would actually split out the functionality:
1. one file for k-medoid
2. one file for featurization
3. one file for distance matrix calculation

Or maybe you could combine 2. and 3.

Couldn't you refactor the existing code so that it works with an iterator, and either pass in a cursor or a list of sections depending on the call site? Copy/pasting is never a good idea...